src/app/delivery-history/delivery-history.component.ts

```diff
@@ -171,17 +182,47 @@ export class DeliveryHistoryComponent implements OnInit {
         let r = d.deliveryStatusDate.isGreaterOrEqualTo(this.dateRange.fromDate.value).and(d.deliveryStatusDate.isLessThan(toDate)).and(d.distributionCenter.filter(this.dialog.distCenter.value))
         if (this.dateRange.onlyDone.value)
           r = r.and(d.deliverStatus.isAResultStatus());
+        if (this.dateRange.onlyArchived.value)
+          r = r.and(d.archive.isEqualTo(true));
         return r;
       }
-    }
+    },
+    rowButtons: [
+      {
+        name: '',
+        icon: 'edit',
+        showInLine: true,
+        visible: () => {return this.settings.usingLabReception.value},
+        click: async fd => {
+          fd.showDetailsDialog({
+            dialog: this.dialog,
+            focusOnDelivery: true
+          });
+        }
+        , textInMenu: () => getLang(this.context).deliveryDetails
+      },
+      {
+        name: '',
+        icon: 'replay',
+        showInLine: true,
+        visible: () => {return (this.settings.usingLabReception.value && this.context.isAllowed(Roles.admin))},
```

Comment:
no need for return - you ca write:
visible; ()=>this.settings.usingLabReception...